fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -813,6 +820,65 @@ public static void analyzeExpression(Expr expression, AnalyzeState state, Scope
         expressionAnalyzer.analyze(expression, state, scope);
     }
 
+    private Type getReturnTypeOfTruncate(FunctionCallExpr node, List<Type> paramsTypes) {
```

Comment:
Incredible!!!!

Reply:
This is unbelievable, do we need to implement the logic of each decimal method in FE?

Reply:
Doesn't decimal have a general rule?

Reply:
This logic has been removed, and now precision and scale of return type are calcuated at be side